assets/js/app.js

```diff
@@ -0,0 +1,51 @@
+const productName = document.getElementById('productName');
+const productPrice = document.getElementById('productPrice');
+const productType = document.getElementById('productType');
+const productImage = document.getElementById('productImage');
+const productForm = document.getElementById('productForm');
+const cartTableBody = document.querySelector('#cartTable tbody');
+
+let products = [];
+
+function addProduct(event) {
+    event.preventDefault();
+    const name = productName.value.trim();
+    const price = parseFloat(productPrice.value.trim());
+    const type = productType.value;
+    const imageUrl = productImage.value.trim();
+
+    if (!name || isNaN(price) || !type || !imageUrl) {
```

Comment:
seems like you're trying to combine HTML and JS validation. please don't. if you need 1 js validation, then turn all the validation to js, otherwise user get's different kind of feedback on various inputs

Reply:
על הוולידציה,גם כאן זה טעות לגמריי, אוליי לא שמתי לב למחוק את זה מהhtml כי הייתי לחוץ על הסוללה. בכל אופן אני מבין לגמריי את השגיאה שלי.